lib/api/v3/work_packages/form/work_package_payload_representer.rb

```diff
@@ -38,6 +38,27 @@ class WorkPackagePayloadRepresenter < Roar::Decorator
           include Roar::JSON::HAL
           include Roar::Hypermedia
 
+          class << self
+            def create_class(work_package)
+              klass = Class.new(WorkPackagePayloadRepresenter)
+              injector_class = ::API::V3::Utilities::CustomFieldInjector
+              property_fields = work_package.available_custom_fields.select do |cf|
+                injector_class.property_field?(cf)
+              end
+
+              injector = injector_class.new(klass)
+              property_fields.each do |custom_field|
+                injector.inject_value(custom_field)
+              end
+
+              klass
```

Comment:
This could probably be moved to the injector, because Version et al will also do something like this.